StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/UsingCodeFixProvider.cs

```diff
@@ -0,0 +1,212 @@
+            var newSyntaxRoot = usingSyntaxRewriter.Visit(syntaxRoot);
+            var indentationOptions = IndentationOptions.FromDocument(document);
+
+            newSyntaxRoot = ProcessUsingDirectives(newSyntaxRoot, usingSyntaxRewriter.UsingDirectives, indentationOptions);
+            var newDocument = document.WithSyntaxRoot(newSyntaxRoot.WithoutFormatting());
+
+            return newDocument;
+        }
+
+        private static SyntaxNode ProcessUsingDirectives(SyntaxNode newSyntaxRoot, ImmutableArray<UsingDirectiveSyntax> usingDirectives, IndentationOptions indentationOptions)
+        {
+            var systemUsings = new List<UsingDirectiveSyntax>();
+            var otherUsings = new List<UsingDirectiveSyntax>();
+            var aliases = new List<UsingDirectiveSyntax>();
+            var staticImports = new List<UsingDirectiveSyntax>();
+
+            foreach (var usingDirective in usingDirectives)
+            {
+                if (usingDirective.Alias != null)
+                {
+                    aliases.Add(usingDirective);
+                }
+                else if (!usingDirective.StaticKeyword.IsKind(SyntaxKind.None))
+                {
+                    staticImports.Add(usingDirective);
+                }
+                else
+                {
+                    var firstPart = usingDirective.Name;
+
+                    while (firstPart is QualifiedNameSyntax)
+                    {
+                        firstPart = ((QualifiedNameSyntax)firstPart).Left;
+                    }
+
+                    var firstPartName = ((IdentifierNameSyntax)firstPart).Identifier.ValueText;
+
+                    if (nameof(System).Equals(firstPartName, StringComparison.Ordinal))
+                    {
+                        systemUsings.Add(usingDirective);
+                    }
+                    else
+                    {
+                        otherUsings.Add(usingDirective);
+                    }
+                }
+            }
+
+            systemUsings.Sort(CompareName);
+            otherUsings.Sort(CompareName);
+            aliases.Sort(CompareAliasName);
+            staticImports.Sort(CompareName);
+
+            var rootNamespace = newSyntaxRoot.ChildNodes().OfType<NamespaceDeclarationSyntax>().FirstOrDefault();
+
+            var indentationLevel = IndentationHelper.GetIndentationSteps(indentationOptions, rootNamespace);
+            var usingsIndentation = IndentationHelper.GenerateIndentationString(indentationOptions, indentationLevel + 1);
+
+            var syntaxList = SyntaxFactory.List<UsingDirectiveSyntax>();
+            syntaxList = AppendUsings(syntaxList, systemUsings, usingsIndentation);
+            syntaxList = AppendUsings(syntaxList, otherUsings, usingsIndentation);
+            syntaxList = AppendUsings(syntaxList, aliases, usingsIndentation);
+            syntaxList = AppendUsings(syntaxList, staticImports, usingsIndentation);
```

Comment:
Could we create one `List<UsingDirectiveSyntax>` with all using in correct order and then create `SyntaxList<UsingDirectiveSyntax>` from it?These mutations looks weird and not effective.